src/AESecb.cpp

```diff
@@ -0,0 +1,144 @@
+/**
+  @file AESecb.cpp
+  Implementation of ECB mode of operation for AES-128 algorithm
+*/
+#include "AESecb.hpp"
+#include <iostream>
+
+
+void printEncryptOutput(std::vector<unsigned char>& output) {
+  std::cout << std::endl;
+  std::cout << "Encrypt result";
+  for (int i = 0; i < output.size() / 16; i++){
+    std::cout << std::endl;
+    std::cout << "Block " << i << ": ";
+    for (int j = 0; j < 16; j++) {
+      std::cout << std::hex << (int) output[j+(i*16)];
+      std::cout << " ";
+    }
+  }
+}
+
+
+/**
+  Implements cipher with ECB mode of operation
+  @param input: array of hex values representing plaintext
+  @param output: array of hex values representing (padded) ciphertext
+  @param key: key to use
+  @param keysize: size of the key
+  @return array of decrypted values
+*/
+void encrypt_ecb(std::array<unsigned char, 16>& input, std::vector<unsigned char>& output, unsigned char* key, unsigned int keysize) {
+
+  // Calculate padding length, then copy input array and padding into plaintext
+  const int inputSize = input.size();
+  const int padLength = 16 - (inputSize % 16);
+  const int plaintextLength = inputSize + padLength;
+
+  std::array<unsigned char, plaintextLength> plaintext;
```

Comment:
Since plaintextLength is variable and not known at compile-time, std::array cannot be used here. This could be a vector instead